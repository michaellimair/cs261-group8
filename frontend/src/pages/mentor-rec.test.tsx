/* eslint-disable no-console */
import { render, RenderResult } from '@testing-library/react';
import { setLogger } from 'react-query';
import { BrowserRouter } from 'react-router-dom';
import MentorRec from './mentor-rec';

describe('mentor-rec', () => {
  let result: RenderResult<typeof import('@testing-library/dom/types/queries'), HTMLElement>;
  let submitButton: HTMLElement;

  beforeAll(() => {
    setLogger({
      log: console.log,
      warn: console.warn,
      // âœ… no more errors on the console
      error: () => {},
    });
  });

  beforeEach(() => {
    result = render(
      <BrowserRouter>
        <MentorRec />
      </BrowserRouter>,
    );
    submitButton = result.queryByTestId('submitButton')!;
    if (!submitButton) {
      throw new Error('Submit button does not exist!');
    }
  });

  afterEach(() => {
    result.unmount();
  });

  it('renders properly', () => {
    expect(result).toMatchSnapshot();
  });
});
