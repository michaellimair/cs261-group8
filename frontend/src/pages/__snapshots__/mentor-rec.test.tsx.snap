// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mentor-rec renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-owjkmg"
      >
        <h2
          class="chakra-heading css-1gjuw9l"
        >
          select_mentors
        </h2>
      </div>
      <div
        class="css-1u9b1ym"
      >
        <button
          class="chakra-button css-5c2176"
          data-testid="submitButton"
          disabled=""
          type="button"
        >
          submit_mentors
        </button>
        <div
          class="chakra-stack css-n21gh5"
        >
          <div
            class="chakra-stack css-dbd5dl"
          >
            <div
              class="chakra-stack css-1tsaady"
            >
              <div
                class="chakra-stack css-1w0t9hc"
              >
                <span
                  class="chakra-avatar css-1ruhx4w"
                >
                  <svg
                    aria-label=" avatar"
                    class="chakra-avatar__svg css-16ite8i"
                    role="img"
                    viewBox="0 0 128 128"
                  >
                    <path
                      d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                      fill="currentColor"
                    />
                    <path
                      d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="chakra-stack css-mjhbcv"
              >
                <h2
                  class="chakra-heading css-1d92c1"
                >
                  Full name 4
                </h2>
                <h2
                  class="chakra-heading css-1d92c1"
                />
                <p
                  class="chakra-text css-13brihr"
                >
                  skill 1, skill 2, skill 3
                </p>
                <p
                  class="chakra-text css-13brihr"
                >
                  England
                </p>
                <p
                  class="chakra-text css-13brihr"
                >
                  Antactica
                </p>
              </div>
              <div
                class="css-l7xm9q"
              >
                <p
                  class="chakra-text css-min50y"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-owjkmg"
    >
      <h2
        class="chakra-heading css-1gjuw9l"
      >
        select_mentors
      </h2>
    </div>
    <div
      class="css-1u9b1ym"
    >
      <button
        class="chakra-button css-5c2176"
        data-testid="submitButton"
        disabled=""
        type="button"
      >
        submit_mentors
      </button>
      <div
        class="chakra-stack css-n21gh5"
      >
        <div
          class="chakra-stack css-dbd5dl"
        >
          <div
            class="chakra-stack css-1tsaady"
          >
            <div
              class="chakra-stack css-1w0t9hc"
            >
              <span
                class="chakra-avatar css-1ruhx4w"
              >
                <svg
                  aria-label=" avatar"
                  class="chakra-avatar__svg css-16ite8i"
                  role="img"
                  viewBox="0 0 128 128"
                >
                  <path
                    d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                    fill="currentColor"
                  />
                  <path
                    d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
            <div
              class="chakra-stack css-mjhbcv"
            >
              <h2
                class="chakra-heading css-1d92c1"
              >
                Full name 4
              </h2>
              <h2
                class="chakra-heading css-1d92c1"
              />
              <p
                class="chakra-text css-13brihr"
              >
                skill 1, skill 2, skill 3
              </p>
              <p
                class="chakra-text css-13brihr"
              >
                England
              </p>
              <p
                class="chakra-text css-13brihr"
              >
                Antactica
              </p>
            </div>
            <div
              class="css-l7xm9q"
            >
              <p
                class="chakra-text css-min50y"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
